src/common/assert.cc

```diff
@@ -33,8 +33,9 @@ namespace ceph {
     g_assert_context = cct;
   }
 
-  void __ceph_assert_fail(const char *assertion, const char *file, int line,
-			  const char *func)
+  [[gnu::cold]] void __ceph_assert_fail(const char *assertion,
```

Comment:
I wasn't aware about this syntax. Much better than poking with `#define` and `__attribute__()`. Thanks!